apps/web/src/services/alert-manager/v2/components/ServiceDetailTabsOverviewStatusTable.vue

```diff
@@ -87,17 +86,35 @@ const state = reactive({
     selectedUrgency: 'ALL',
 });
 const tableState = reactive({
-    loading: false,
     alertsList: [] as AlertModel[],
     alertStateLabels: getAlertStateI18n(),
     urgencyLabels: getAlertUrgencyI18n(),
-    totalCounts: 0,
     sortBy: '',
     sortDesc: false,
 });
 
 const alertListApiQueryHelper = new ApiQueryHelper().setSort('created_at', true)
     .setPage(1, 15);
+const { alertAPI } = useAlertApi();
+const { key: alertListQueryKey, params: alertListQueryParams } = useServiceQueryKey('alert-manager', 'alert', 'list', {
+    params: computed(() => ({
+        service_id: storeState.serviceInfo.service_id,
+        status: state.selectedStatus,
+        urgency: state.selectedUrgency === 'ALL' ? undefined : state.selectedUrgency as AlertUrgencyType,
+        query: alertListApiQueryHelper.data,
+    })),
+});
+const { data: alertListData, isLoading: alertListLoading, refetch: refetchAlertList } = useQuery({
+    queryKey: alertListQueryKey.value,
+    queryFn: async () => {
+        const res = await alertAPI.list(alertListQueryParams.value);
+        return {
+            results: res.results ?? [],
+            totalCount: res.total_count ?? 0,
+        };
+    },
+    enabled: false,
+});
```

Comment:
Suggestion: Rather than using refetch directly, it is recommended that useQuery be enabled declaratively whenever the variable bound to queryKey params changes.

---
직접적으로 refetch를 사용하는 것보다 queryKey params에 바인딩된 변수가 바뀔 때마다 선언적으로 useQuery가 작동되게 하는 것이 좋습니다.
이렇게 작성하신 다른 이유가 있을까요?

Reply:
- If you need to invalidate it elsewhere with the corresponding queryKey (with reactive params applied), it's also a good idea to separate it into a compatible one!
---
해당하는 queryKey(반응성을 가진 params가 적용된)로 다른 곳에서 무효화를 진행해야한다면, queryKey의 재사용을 위해 composable로 분리하는 것도 좋은 방법입니다!